src/instructlab/sdg/filterblock.py

```diff
@@ -11,25 +11,48 @@
 
 class FilterByValueBlock(Block):
     def __init__(
-        self, filter_column, filter_value, operation, convert_dtype=None, **batch_kwargs
+        self,
+        filter_column,
+        filter_value,
+        operation,
+        valid_values,
```

Comment:
Something that would help a lot here is adding some documentation of these parameters that explains how they're all to be used. That would, I hope, help some of the discussions we're having trying to understand the changes. Can you add a docstring to this method?

Reply:
https://github.com/instructlab/sdg/issues/76 Sounds good, lets address this in a follow up

Reply:
Yeah, one thing that helps is ... `filter_column`, `filter_value`, and `operation` go together

`filter_column` and `valid_values` go together too

so, it's like:

- filter on filter_column,
- first (and optionally) by checking the column contains one of valid_values
- second by comparing the value in the column matches filter_value using the given operation

Reply:
and the confusion is "why do we need the second bullet again?"  right?

is it that actually you really only need one or the other?

or what's the example for when you need them both together?

Reply:
... or put another way, two types of filtering?

could they be collapsed into one, where it's always a set, and you might just have only one element in that set?